btree/btree.h

```diff
@@ -120,6 +141,25 @@ void m0_btree_op_credit(const struct m0_btree_op *bt,
 int  m0_btree_mod_init(void);
 void m0_btree_mod_fini(void);
 
+
+/**
+ * This macro calls the 'action' function and follows it by calling
+ * m0_sm_op_tick() to execute the state machine.
+ *
+ * IMPORTANT: The 'action' routine should execute the call m0_sm_op_init() for
+ * setting the *_tick() function which is eventually be called by
+ * m0_sm_op_tick() function.
+ */
+#define M0_BTREE_OP_SYNC_WITH(op, action)       \
+	({                                      \
+		struct m0_sm_op *__opp = (op);  \
+						\
+		action;                         \
+		m0_sm_op_tick(__opp);           \
+		m0_sm_op_fini(__opp);           \
```

Comment:
I think we might need to consider following functions as a part of this macro:

m0_thread_exec_fini()
m0_sm_group_unlock()

Since we are using state machine approach, these functions should be called after m0_sm_op_fini(); to finish the respective state machine operations.

Reply:
Same suggestion as above.